packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -274,55 +284,232 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
+        (variable) => variable.boundVar.name == 'this',
+        orElse: () => null);
+    if (thisVariable != null) {
+      final thisValue = thisVariable.boundVar.value;
+      if (thisValue is InstanceRef) {
+        await _addAllInstanceMembersToAutocompleteList(
+          result,
+          thisValue,
+          controller,
+        );
+      }
+    }
+    final frame = controller.frameForEval;
+    if (frame != null) {
+      final function = frame.function;
+      if (function != null) {
+        final libraryRef = await findOwnerLibrary(function, controller);
+        if (libraryRef != null) {
+          controller.autocompleteInclusiveCache.clear();
+          controller.autocompleteCache.clear();
+          result.addAll(await libraryMemberAndImportsAutocompletes(
+              libraryRef, controller));
+        }
+      }
+      // TODO(jacobr): consider
+    }
   } else {
     var left = parts.leftSide.split(' ').last;
     // Removing trailing `.`.
     left = left.substring(0, left.length - 1);
     try {
       final response = await controller.evalAtCurrentFrame(left);
       if (response is InstanceRef) {
-        final Instance instance = await controller.getObject(response);
-        result.addAll(
-          await _autoCompleteMembersFor(
-            instance.classRef,
+        if (response.typeClass != null) {
+          // Assume we want static members for a type class not members of the
+          // Type object. This is reasonable as Type objects are rarely useful
+          // in dart.
+          result.addAll(await _autoCompleteMembersFor(
+            response.typeClass,
             controller,
-          ),
-        );
-        // TODO(grouma) - This shouldn't be necessary but package:dwds does
-        // not properly provide superclass information.
-        final clazz = await classFor(instance.classRef, controller);
-        result.addAll(instance.fields
-            .map((field) => field.decl.name)
-            .where((member) => _isAccessible(member, clazz, controller)));
+            staticContext: true,
+          ));
+        } else {
+          await _addAllInstanceMembersToAutocompleteList(
+            result,
+            response,
+            controller,
+          );
+        }
       }
     } catch (_) {}
   }
-  return result.where((name) => name.startsWith(parts.activeWord)).toList()
-    ..sort();
+  return result.where((name) => name.startsWith(parts.activeWord)).toList();
 }
 
-Future<Set<String>> _autoCompleteMembersFor(
-  ClassRef classRef,
+// Due to https://github.com/dart-lang/sdk/issues/46221
+// we cannot tell what the show clause for an export was so it is unsafe to
+// surface exports as if they were library members as there tend to be
+// significant false positives for libraries such as Flutter where all of
+// dart:ui shows up as in scope from flutter:foundation when it should not be.
+bool debugIncludeExports = true;
+
+Future<Set<String>> libraryMemberAndImportsAutocompletes(
```

Comment:
Not clear to me why this needs to be public.

Reply:
> This PR really needs a description of what it's trying to achieve. There's a lot to this polish.

done.